src/main/java/com/github/manolo8/darkbot/config/Config.java

```diff
@@ -312,33 +312,61 @@ public static class ShipConfig implements LegacyShipMode {
         public int CONFIG = 1;
         public Character FORMATION;
 
+        private HeroAPI.Configuration newConfiguration;
+        private SelectableItem.Formation newFormation;
+
         public ShipConfig() {}
+
+        public ShipConfig(HeroAPI.Configuration configuration, SelectableItem.Formation formation) {
+            this.newConfiguration = configuration;
+            this.newFormation = formation;
+        }
+
         public ShipConfig(int CONFIG, Character FORMATION) {
             this.CONFIG = CONFIG;
             this.FORMATION = FORMATION;
         }
 
         @Override
         public HeroAPI.Configuration getConfiguration() {
-            return HeroAPI.Configuration.of(CONFIG);
+            if (CONFIG != -1) {
+                newConfiguration = HeroAPI.Configuration.of(CONFIG);
+                CONFIG = -1;
```

Comment:
for ideal backwards compatibility, you'd want to keep CONFIG and FORMATION updated too, just use the new ones if available preferrably

Reply:
i think the issue will be, since the standard formation is being set by default, it will likely use the "NEW" standard and not the one set by the key or update?

Reply:
wait not sure where to keep the old values updated from, i think from ShipModeEditor?